torchvision/datasets/utils.py

```diff
@@ -42,6 +42,22 @@ def check_integrity(fpath: str, md5: Optional[str] = None) -> bool:
     return check_md5(fpath, md5)
 
 
+def _get_redirect_url(url: str, hop_idx: int = 0, max_hops: int = 10) -> str:
```

Comment:
Given that a user probably won't use this function as a standalone, we should propagate the `max_hops` parameter to `download_url`. In there I suggest we rename it to `max_redirects` or `max_redirect_hops` to make its purpose more clear.